src/main/java/org/sonarlint/intellij/util/FutureUtils.java

```diff
@@ -41,12 +44,34 @@ public static void waitForTask(Project project, Future<?> task, String taskName,
     }
   }
 
-  public static void waitForTasks(Project project, List<Future<?>> updateTasks, String taskName) {
+  public static void waitForTasks(Project project, ProgressIndicator indicator, List<Future<?>> updateTasks, String taskName) {
     for (var f : updateTasks) {
-      waitForTask(project, f, taskName, Duration.ofSeconds(20));
+      waitForTask(project, indicator, f, taskName, Duration.ofSeconds(20));
     }
   }
 
+  private static void waitForFuture(ProgressIndicator indicator, Future<?> future, Duration durationTimeout) throws InterruptedException, ExecutionException, TimeoutException {
```

Comment:
You can probably make `FutureAwaitingTask` use this method

Reply:
Not sure, yours is running indefinitely

Reply:
true